### ISSUE TITLE
j.u.regex returns wrong group

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#1506]
re2s is going to need another coup stick.

On Scala JVM the sequence below return the value expected by visual execution
of the pattern "bcde".  This is also the pattern expected by re2s RE2Matcher find
tests.

On SN, the value returned is: "abcde". That is, the start index is ignored.

Sn 
```
scala> val text = "abcdefgh"
scala> val regexp = ".*[aeiou]"
scala>  val pj           = java.util.regex.Pattern.compile(regexp)
scala>     val matchStringj = pj.matcher(text)
scala> matchStringj.find(1)
scala> matchStringj.group
res1: String = bcde

```

**Comment by LeeTibbert** [comment]
Too early to close yet, but looks like this will be fixed when j.u.regex uses re2s.

**Comment by LeeTibbert** [comment]
As of the merged PR #1600, j.u.regex.MatcherSuite.scala has a test case for this Issue:
`test("find(start) - group") { // As reported in Issue #1506 `

That test case has been executing & passing for slightly over a month now, so I am closing
this Issue as Fixed.
